hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/Contexts.java

```diff
@@ -108,26 +131,78 @@ public ILogger logger() {
         public ProcessingGuarantee processingGuarantee() {
             return processingGuarantee;
         }
+
     }
 
     static class ProcSupplierCtx extends MetaSupplierCtx implements ProcessorSupplier.Context {
 
         private final int memberIndex;
+        private final ConcurrentMap<String, File> localFiles;
 
         @SuppressWarnings("checkstyle:ParameterNumber")
         ProcSupplierCtx(
-                JetInstance jetInstance, long jobId, long executionId, JobConfig jobConfig, ILogger logger,
-                String vertexName, int localParallelism, int totalParallelism, int memberIndex, int memberCount,
-                ProcessingGuarantee processingGuarantee) {
+                JetInstance jetInstance,
+                long jobId,
+                long executionId,
+                JobConfig jobConfig,
+                ILogger logger,
+                String vertexName,
+                int localParallelism,
+                int totalParallelism,
+                int memberIndex,
+                int memberCount,
+                ProcessingGuarantee processingGuarantee,
+                ConcurrentMap<String, File> localFiles
+        ) {
             super(jetInstance, jobId, executionId, jobConfig, logger, vertexName, localParallelism, totalParallelism,
                     memberCount, processingGuarantee);
             this.memberIndex = memberIndex;
+            this.localFiles = localFiles;
         }
 
         @Override
         public int memberIndex() {
             return memberIndex;
         }
+
+        @Nonnull @Override
+        public File attachedDirectory(@Nonnull String id) {
+            Preconditions.checkHasText(id, "id cannot be null or empty");
+            findResourceConfigOrThrowException(id);
```

Comment:
Why don't we just rely on the presence of the key in the file IMap? This is superfluous. I was even thinking whether we shouldn't make the `resourceConfigs` transient in `JobConfig`, the URLs might be invalid on another member. The ID is enough and the user is supposed to know the IDs, we don't need to have them on members.